packages/core/src/state-management/state-manager.ts

```diff
@@ -191,7 +191,8 @@ export class StateManager {
     const shouldIndex =
       state$.state.metadata.model && this._index.shouldIndexStream(state$.state.metadata.model)
     if (isPinned || shouldIndex) {
-      await this.pinStore.add(state$)
+      // force: true to save the updated stream in the stream state store, even if there hadn't been any new pinned commits
```

Comment:
this will cause a lot of extra work to be done to re-pin every IPFS record in the entire stream log.  Is this to work around the issue where we don't update the state in the state store if the only thing that changed is the anchorStatus?  If so I think we need a more targeted fix for that.  Does that need to be fixed in this PR or could that be a follow-up?  Is it something we even need to worry about fixing right now?